package.json

```diff
@@ -18,7 +18,9 @@
     "test": "npm run build && npm run lint && mocha -r ts-node/register test/index.spec.ts",
     "build": "tsc",
     "lint": "tslint -t stylish -c tslint.json 'src/**/*.ts' 'test/**/*.ts'",
-    "docs": "npm run build && node generateDocs"
+    "docs": "npm run build && node generateDocs",
+    "precommit": "lint-staged",
+    "prettify": "prettier --write \"src/**/*.ts\" \"tests/**/*.ts\""
```

Comment:
As mentioned in the [other review](https://github.com/resin-io-modules/resin-semver/pull/48#discussion_r138876625): personally I'd love to have `npm run build` be a reliable step to do all compilation steps on every project, and I think most of our other npm projects do that (and the others really should). What do you think? Could we trigger docs, prettier and linting in there too, so the lint-staged hook just has to run `npm run build`?

If there's downsides to that because `tsc` is faster than the other steps and it's annoying, I've got a few places where I've been adding `build:<step>` jobs for the parts you might want separately, so you can manually trigger them individually when that's useful. If you want to run tsc alone often, having `build:tsc` step that's run by `build` sounds fine to me, it's just that I'd love reliable `npm run build` triggers that build everything everywhere.

Reply:
agreed 💯 I'll address this issue in the other PR